lib/chef/provider/ifconfig/redhat.rb

```diff
@@ -22,6 +22,7 @@ class Chef
   class Provider
     class Ifconfig
       class Redhat < Chef::Provider::Ifconfig
+        provides :ifconfig, platform: %w(centos fedora redhat ibm_powerkvm cloudlinux parallels)
```

Comment:
yes to both.  and you're missing oracle and scientific, both of which should be here, which is why platform_family is better.